### PR TITLE
Explicitly set dir mode to 0775. Fix malformed license metadata.

### PR DESCRIPTION
**Pull request by ZebraFlesh**
Fixes ember-cli/ember-cli#4813

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -156,7 +156,7 @@ function writeP(filePath, content) {
 
   return writeFile(filePath, content).catch(function(reason) {
     if (reason && reason.code === 'ENOENT') {
-      return mkdirp(base).then(function() {
+      return mkdirp(base, { mode: '0775' }).then(function() {
```
Comment (stefanpenner) [inline index.js]: seems like we shouldn't give anyone execute, just read/write correct ?

Comment (mike-post) [inline index.js]: No, you need write+execute on directories if you actually want to modify the contents.  See  http://www.hackinglinuxexposed.com/articles/20030424.html

Comment (stefanpenner) [inline index.js]: ah thanks!

**Comment by stefanpenner** [comment]
released as v1.0.3 – tested in an app, all appears well.

**Comment by MSch** [comment]
I'm still getting this error with the latest ember-cli with async-disk-cache 1.0.3 in my `npm-shrinkwrap.json`.

**Comment by ef4** [comment]
I don't think `775` is sufficient. It's still going to troll users, because it doesn't help unless you're manually managing group membership, on both the top level and second level directories. And reliably creating second-level directories with the right group would require something like the set-group-id bit too.

`/tmp` itself uses `1777`, which is closer to what we want. (The `1` means sticky, meaning you can only unlink things from `/tmp` if you are their owner.) But even `1777` will be annoying if you ever need to clear the cache, because you'll only be able to remove files and second-level directories that you created.

The only works-out-of-the-box, non-annoying option is `777` for the directories. The downside being that any random unprivileged process can start messing with your cache.

I think it will be ultimately less painful to use `/tmp/async-cache-$USERNAME` instead. A shared cache is theoretically nice, but in practice will be pretty annoying in shared environments where not everyone is a superuser.

**Comment by ef4** [comment]
Oh, I notice that we're already using `666` for files, so any security argument against `777` for the directories is kinda moot. If you don't like `777` on dirs, you also don't want to keep `666` on files.

**Pull request merged**